### PR TITLE
Clean up new webapp

### PR DESCRIPTION
**Pull request by rainy789**
## Motivation
Clean up and fix multiple issues. 

## Modification
- Clean up link button. Fix the issue when `Application tokens` was  occasionally not redirecting. 
Screen Shot 2023-02-22 at 12 03 03

- Update linkable tabs from hash to query param. Fix the cancel rendering route error issue when changing tabs too quickly. 

- Add linkable tabs for repo permission page.

- Use a generalised `Deferred` for `error` and `isLoading`. 
- Add a spinner. 

- Display a generic error page instead of raw json. Address [https://github.com/line/centraldogma/pull/796#discussion_r1099786793](https://github.com/line/centraldogma/pull/796#discussion_r1099786793)

- Exclude deleted projects from the jump dropdown as we won't be able to jump to the project. 

- Rename `ready` to `isLoading`. Hopefully it's less confusing. Address https://github.com/line/centraldogma/pull/760#discussion_r1030093920

- When navigating to an unknown page, the 404 page looks broken in light mode. Add a customised 404 page.

## Result
- `npm run develop`

### Linkable tabs
Screen Shot 2023-02-22 at 12 05 06

### Generic error page - light
Screen Shot 2023-02-17 at 14 19 19

### Generic error page - dark
Screen Shot 2023-02-17 at 13 44 17

### Dropdown with active projects only
Screen Shot 2023-02-22 at 12 57 03

#### before
Screen Shot 2023-02-22 at 13 45 41

#### after
Screen Shot 2023-02-22 at 13 53 00

### CODE REVIEW

File: webapp/src/pages/app/projects/metadata/[projectName]/index.tsx [hunk 1]
```diff
@@ -23,90 +24,103 @@ const ProjectMetadataPage = () => {
...
+            <Heading size="lg">Project {projectName} - Metadata</Heading>
+          </Flex>
+          <Tabs variant="enclosed-colored" size="lg" index={tabIndex}>
+            <TabList>
+              {tabs.map((tabName) => (
+                <Tab
+                  as={Link}
+                  key={tabName}
+                  replace
+                  href={{ pathname: `/app/projects/metadata/${projectName}`, query: { tab: tabName } }}
```
Comment (ikhoon) [inline webapp/src/pages/app/projects/metadata/[projectName]/index.tsx:62]: 👍

**Review by ikhoon — APPROVED** [review]
Nice cleanup! 🧹🙇‍♂️

**Comment by ikhoon** [comment]
I forget to merge this PR. 😅

**Comment by inch772** [comment]
mantap~ @rainy789

**Pull request merged**